spec.bs

```diff
@@ -466,12 +560,18 @@ An {{AppHistoryNavigateEvent}} also has an associated {{Promise}}-or-null <dfn f
      <p class="note">This can occurr if an event listener disconnected the <{iframe}> corresponding to [=this=]'s [=relevant global object=].</p>
   1. If |event|'s [=AppHistoryNavigateEvent/navigation action promise=] is non-null, or both |result| and |isSameDocument| are true, then:
     1. If |event|'s [=AppHistoryNavigateEvent/navigation action promise=] is null, then set it to [=a promise resolved with=] undefined, created in |realm|.
+    1. Let |navigateMethodCallPromise| be [=this=]'s [=AppHistory/navigate method call promise=].
```

Comment:
Putting all these steps into "fire a navigate event" (instead of in the navigate algorithm itself) is starting to feel a bit strange. But it is all very app history related. We'll see how things develop as we do state and event.destination.